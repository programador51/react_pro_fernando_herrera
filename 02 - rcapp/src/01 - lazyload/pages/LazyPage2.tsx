import React from "react";

export default function LazyPage() {
  return <h1>Lazy Page 2</h1>;
}
