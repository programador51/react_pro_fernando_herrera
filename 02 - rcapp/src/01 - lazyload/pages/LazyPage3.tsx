import React from "react";

export default function LazyPage() {
  return <h1>Lazy Page 3</h1>;
}
