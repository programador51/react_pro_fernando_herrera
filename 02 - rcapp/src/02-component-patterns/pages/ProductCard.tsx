import React, { useState } from "react";
import styles from "../styles/styles.module.css";
import useProduct from "02-component-patterns/hooks/useProduct";

interface Props {
  product: Product;
}

interface Product {
  id: string;
  title: string;
  img?: string;
}

export default function ProductCard({ product }: Props) {
  const { counter, decrease, increase } = useProduct();

  console.log(product.img);

  const img =
    product.img !== undefined
      ? product.img
      : `${process.env.PUBLIC_URL}/images/no-image.jpg`;

  return (
    <div className={styles.productCard}>
      <img
        className={styles.productImg}
        src={img}
        alt="mug of the product to sell"
      />
      <span className={styles.productDescription}>{product.title}</span>

      <div className={styles.buttonsContainer}>
        <button onClick={decrease} className={styles.buttonMinus}>
          -
        </button>
        <div className={styles.countLabel}>{counter}</div>
        <button onClick={increase} className={styles.buttonAdd}>
          +
        </button>
      </div>
    </div>
  );
}
