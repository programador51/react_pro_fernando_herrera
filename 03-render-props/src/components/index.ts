import CheckboxInput from "./CheckboxInputs";
import SelectInput from "./SelectInput";
import TextInput from "./TextInput";

export {
    CheckboxInput, SelectInput, TextInput
}