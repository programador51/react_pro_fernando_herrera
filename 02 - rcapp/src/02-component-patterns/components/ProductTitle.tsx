import { ProductContext } from "02-component-patterns/pages/ProductCard";
import { CSSProperties, useContext } from "react";
import styles from "../styles/styles.module.css";

export interface Props {
  title?: string;
  className?: string;
  style?: CSSProperties;
}

export const ProductTitle = ({ title, className = "", style = {} }: Props) => {
  const { product } = useContext(ProductContext);

  return (
    <span style={style} className={`${styles.productDescription} ${className}`}>
      {title ? title : product.title}
    </span>
  );
};
