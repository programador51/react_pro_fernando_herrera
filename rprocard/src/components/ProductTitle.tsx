import React from 'react';
import { ProductContext } from '../pages/ProductCard';
import { CSSProperties, useContext } from 'react';
import styles from '../styles/styles.module.css';

export interface PropsProductTitle {
  title?: string;
  className?: string;
  style?: CSSProperties;
}

export const ProductTitle = ({
  title,
  className = '',
  style = {},
}: PropsProductTitle) => {
  const { product } = useContext(ProductContext);

  return (
    <span style={style} className={`${styles.productDescription} ${className}`}>
      {title ? title : product.title}
    </span>
  );
};
