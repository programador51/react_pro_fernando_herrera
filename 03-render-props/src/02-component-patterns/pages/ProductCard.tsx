import React, { createContext } from "react";
import styles from "../styles/styles.module.css";
import useProduct from "02-component-patterns/hooks/useProduct";
import {
  ProductContextProps,
  Props,
} from "02-component-patterns/interfaces/products.interfaces";
import { ProductImage } from "02-component-patterns/components/ProductImage";
import { ProductTitle } from "02-component-patterns/components/ProductTitle";
import { ProductButtons } from "02-component-patterns/components/ProductButtons";

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export function ProductCard({
  product,
  children,
  className = "",
  onChange = () => {},
  value = 0,
  initialValues,
}: Props) {
  const { counter, decrease, increase } = useProduct({
    onChange,
    product,
    value,
    initialValues,
  });

  return (
    <Provider
      value={{
        counter,
        decrease,
        increase,
        product,
      }}
    >
      <div className={`${styles.productCard} ${className}`}>
        {children("hi")}
      </div>
    </Provider>
  );
}

ProductCard.Title = ProductTitle;
ProductCard.Image = ProductImage;
ProductCard.Buttons = ProductButtons;
