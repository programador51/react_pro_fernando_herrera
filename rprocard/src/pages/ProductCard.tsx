import React, { createContext } from 'react';
import styles from '../styles/styles.module.css';
import useProduct from '../hooks/useProduct';
import { ProductContextProps, Props } from '../interfaces/products.interfaces';
import { ProductImage } from '../components/ProductImage';
import { ProductTitle } from '../components/ProductTitle';
import { ProductButtons } from '../components/ProductButtons';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export function ProductCard({
  product,
  children,
  className = '',
  onChange = () => {},
  value = 0,
  initialValues,
}: Props) {
  const {
    counter,
    decrease,
    increase,
    maxCount,
    reset,
    isMaxQuantityReached,
  } = useProduct({
    onChange,
    product,
    value,
    initialValues,
  });

  return (
    <Provider
      value={{
        counter,
        decrease,
        increase,
        product,
        maxCount,
      }}
    >
      <div className={`${styles.productCard} ${className}`}>
        {children({
          quantity: counter,
          isMaxQuantityReached,
          maxQuantity: initialValues?.maxQuantity,
          product,
          increase,
          decrease,
          reset,
        })}
      </div>
    </Provider>
  );
}

ProductCard.Title = ProductTitle;
ProductCard.Image = ProductImage;
ProductCard.Buttons = ProductButtons;
