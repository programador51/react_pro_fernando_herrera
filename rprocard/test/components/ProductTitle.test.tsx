import React from 'react';
import render from 'react-test-renderer';
import { ProductTitle } from '../../src/components/ProductTitle';
import { ProductCard } from '../../src/pages/ProductCard';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('Debe de mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper = render.create(<ProductTitle title="Custom product" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Mostrar el componente con el nombre del producto', () => {
    const wrapper = render.create(
      <ProductCard product={product1}>
        {() => (
          <>
            <ProductTitle />
          </>
        )}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
