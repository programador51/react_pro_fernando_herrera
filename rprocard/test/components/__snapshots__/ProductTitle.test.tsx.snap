// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription "
  style={Object {}}
>
  Custom product
</span>
`;

exports[`ProductTitle Mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard "
>
  <span
    className="productDescription "
    style={Object {}}
  >
    Taza Dev
  </span>
</div>
`;
